design/specs/specs-null-values.md

```diff
@@ -0,0 +1,128 @@
+
+## User Experience
+
+### Considerations for handling NULL Values
+
+When presented in tables, NULL values should be visually distinct so that the user understands that they are not the same as other string types. The style should attempt to communicate the absence of a value, similar to how placeholders are used in text input fields.
+As for the process of transforming values to and from NULL, it will be important to have a consistent way of doing this across different data types.
+
+## Scenarios
+
+### Scenario 1: User identifies a NULL Value
+
+#### Scenario 1a: The user identifies a NULL value in a table
+
+##### Steps for 1a
+
+1. The user opens a table that contains NULL values
+2. The user identifies a cell that contains a NULL value
+    - The cell contains the word NULL in all uppercase letters
+    - The cell has a lighter background-color
+    - The cell has a muted text color
+    - The cell has a different font style (italic)
+
+#### Scenario 1b: The user identifies a NULL value in a record preview
+
+##### Steps for 1b
+
+1. The user edits a value in a column that links to records from another table
+2. The user sees the list of records displayed in the record selection dropdown
+3. The user identifies a record that contains a NULL value
+    - The field contains the word NULL in all uppercase letters
+    - The text has a muted text color
+    - The text has a different font style (italic)
+
+### Scenario 2: User edits a NULL Value
+
+#### Scenario 2a: The data type is text
+
+##### Steps for 2a
+
+1. The user edits a NULL value in a column that is set to text data type
+2. The user double-clicks the value cell to enter edit mode
+3. The user enters a value and leaves edit mode by pressing the enter key or clicking outside of the active cell
+4. The NULL value is replaced by the new value
+
+#### Scenario 2b: The data type is text, and the value is cleared
+
+#### Steps for 2b
+
+1. The user edits a NULL value in a column that is set to text data type
+2. The user double-clicks the value cell to enter edit mode
+3. In edit-mode, a text input box is displayed with a placeholder set to 'NULL'
+4. The user types in a value and the placeholder text is replaced by the new typed-in value
+5. The user clears the new value by hitting the backspace key until the value is completely cleared
+6. The user exits edit-mode
+7. The NULL value is replaced by an empty string and the cell is blank
+
+#### Scenario 2c: The data type is boolean with checkbox display enabled
+
+##### Steps for 2c
+
+1. The user edits a NULL value in a column that is set to boolean type with the checkbox display property enabled
+    - The field contains a checkbox in an indeterminate state
+    - Hovering over the NULL checkbox will display a tooltip with the 'NULL' legend.
+2. The user clicks on the checkbox, going from indeterminate (NULL) to checked (TRUE).
+3. The user can click on the checkbox again to mark it as false and go back to NULL.
```

Comment:
I'm not sure that we should support going back to `NULL` just by clicking on it. The general pattern with checkboxes is that clicking toggles TRUE/FALSE and I don't think we should violate that. I think we should only allow them to set this to NULL using the contextual menu.

Reply:
Agree with kgodey, `null` should be seen as a state of the row and not as a value of a row., so an interface to change the value of a row shouldn't also change the state of the row.